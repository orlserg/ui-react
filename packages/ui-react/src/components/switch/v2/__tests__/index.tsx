import React from 'react';
import { create } from 'react-test-renderer';

import { Switch } from '..';

describe('Default', () => {
    test('Not checked', () => {
        const tree = create(
            <Switch
                name="default"
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('With ID', () => {
        const tree = create(
            <Switch
                id="custom-id"
                name="default"
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Switch
                name="default"
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Switch
                name="default"
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Switch
                name="default"
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe('Label', () => {
    const PROPS = {
        name : 'default',
        label: 'Checkbox label'
    };

    test('Not checked', () => {
        const tree = create(
            <Switch
                {...PROPS}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Switch
                {...PROPS}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Switch
                {...PROPS}
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Switch
                {...PROPS}
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe('Label & Description', () => {
    const PROPS = {
        name       : 'default',
        label      : 'Checkbox label',
        description: 'Checkbox description'
    };

    test('Not checked', () => {
        const tree = create(
            <Switch
                {...PROPS}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Switch
                {...PROPS}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Switch
                {...PROPS}
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Switch
                {...PROPS}
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
