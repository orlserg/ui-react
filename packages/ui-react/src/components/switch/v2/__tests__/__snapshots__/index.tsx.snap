// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Checked & disabled 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    />
  </button>
</label>
`;

exports[`Default Checked & loading 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    >
      <span
        className="switch__track-point_loading"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 7H6.193C7.553 5.079 9.606 4 12 4a8 8 0 0 1 8 8h2c0-5.523-4.477-10-10-10-2.804 0-5.275 1.183-7 3.27V2H3v7h7V7Zm4 10h3.807c-1.36 1.921-3.413 3-5.807 3a8 8 0 0 1-8-8H2c0 5.523 4.477 10 10 10 2.804 0 5.275-1.183 7-3.27V22h2v-7h-7v2Z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </button>
</label>
`;

exports[`Default Checked 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked"
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    />
  </button>
</label>
`;

exports[`Default Not Checked & loading 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point"
    >
      <span
        className="switch__track-point_loading"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 7H6.193C7.553 5.079 9.606 4 12 4a8 8 0 0 1 8 8h2c0-5.523-4.477-10-10-10-2.804 0-5.275 1.183-7 3.27V2H3v7h7V7Zm4 10h3.807c-1.36 1.921-3.413 3-5.807 3a8 8 0 0 1-8-8H2c0 5.523 4.477 10 10 10 2.804 0 5.275-1.183 7-3.27V22h2v-7h-7v2Z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </button>
</label>
`;

exports[`Default Not checked & disabled 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  >
    <div
      className="switch__track-point"
    />
  </button>
</label>
`;

exports[`Default Not checked 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track"
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point"
    />
  </button>
</label>
`;

exports[`Default With ID 1`] = `
<label
  className="switch switch_size-md"
  id="custom-id"
>
  <button
    aria-checked={false}
    className="switch__track"
    id="custom-id-input"
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point"
    />
  </button>
</label>
`;

exports[`Label Checked & disabled 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    />
  </button>
  Checkbox label
</label>
`;

exports[`Label Checked & loading 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    >
      <span
        className="switch__track-point_loading"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 7H6.193C7.553 5.079 9.606 4 12 4a8 8 0 0 1 8 8h2c0-5.523-4.477-10-10-10-2.804 0-5.275 1.183-7 3.27V2H3v7h7V7Zm4 10h3.807c-1.36 1.921-3.413 3-5.807 3a8 8 0 0 1-8-8H2c0 5.523 4.477 10 10 10 2.804 0 5.275-1.183 7-3.27V22h2v-7h-7v2Z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </button>
  Checkbox label
</label>
`;

exports[`Label Checked 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={true}
    className="switch__track switch__track_checked"
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point switch__track-point_checked"
    />
  </button>
  Checkbox label
</label>
`;

exports[`Label Not Checked & loading 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point"
    >
      <span
        className="switch__track-point_loading"
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 7H6.193C7.553 5.079 9.606 4 12 4a8 8 0 0 1 8 8h2c0-5.523-4.477-10-10-10-2.804 0-5.275 1.183-7 3.27V2H3v7h7V7Zm4 10h3.807c-1.36 1.921-3.413 3-5.807 3a8 8 0 0 1-8-8H2c0 5.523 4.477 10 10 10 2.804 0 5.275-1.183 7-3.27V22h2v-7h-7v2Z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </button>
  Checkbox label
</label>
`;

exports[`Label Not checked & disabled 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track switch__track_disabled"
    disabled={true}
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  >
    <div
      className="switch__track-point"
    />
  </button>
  Checkbox label
</label>
`;

exports[`Label Not checked 1`] = `
<label
  className="switch switch_size-md"
>
  <button
    aria-checked={false}
    className="switch__track"
    name="default"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <div
      className="switch__track-point"
    />
  </button>
  Checkbox label
</label>
`;
